pkg/health/checker/checker.go

```diff
@@ -265,23 +345,110 @@ func (c consulHealthChecker) WatchService(
 				case resultCh <- out:
 				}
 			}
+			if useHealthService {
+				// get status endpoints for service
+				statusEndpoints, err := h.getStatusEndpoints(serviceID, status)
+				if err != nil {
+					select {
+					case <-quitCh:
+						return
+					case errCh <- err:
+					}
+					continue
+				}
+				for _, endpoint := range statusEndpoints {
+					healthReq := &hclient.HealthRequest{
```

Comment:
Does this need to change given the new health service proto? It's kind of annoying that the protos are closed source - it creates a bit of a circular dependency.

Reply:
This doesn't need to change since this is calling HealthMonitor and it still uses HealthRequest messages.